uml4net/POCO/StructuredClassifiers/Class.cs

```diff
@@ -409,6 +409,6 @@ public class Class : XmiElement, IClass
         [Property(aggregation: AggregationKind.None, lowerValue: 0, upperValue: int.MaxValue, isDerived: true)]
         [RedefinedProperty("Class-superClass")]
         [Implements(implementation: "IClass.SuperClass")]
-        public List<IClass> SuperClass => throw new NotImplementedException();
+        public List<IClass> SuperClass => this.Generalization.Select(x => x.General).OfType<IClass>().ToList();
```

Comment:
we leave this for now, but i suspect that when we code generate this will be overriden by the code generator. A pattern to handle this is requried.

I propose we make all classes "partial" and hand code these algorithms in the extension where we follow the following convention (i hope the example below makes it explicit)

```
public List<IClass> SuperClass =>this.QuerySuperClass();
```

we prefix the word `Query` in front of each method that implements the derived (isDerived:true) property and each of these is handcoded either in a partial class, or in a Static Class that provides extension methods.